src/task/future.rs

```diff
@@ -151,6 +172,47 @@ unsafe fn clone_task(task: *const Task) -> TaskCell {
     task_cell
 }
 
+thread_local! {
+    /// Local queue reference that is set before polling and unset after polled.
+    static LOCAL: Cell<*mut Local<TaskCell>> = Cell::new(std::ptr::null_mut());
+}
+
+unsafe fn wake_task(task: Cow<'_, Arc<Task>>, reschedule: bool) {
+    LOCAL.with(|ptr| {
+        if ptr.get().is_null() {
+            // It's out of polling process, has to be spawn to global queue.
+            // It needs to clone to make it safe as it's unclear whether `self`
+            // is still used inside method `spawn` after `TaskCell` is dropped.
+            (*task.as_ref().extras.get())
+                .remote
+                .as_ref()
+                .expect("waker should not be moved to other threads by reference")
+                .spawn(TaskCell(task.clone().into_owned()));
```

Comment:
Sorry I still don't understand why we need a clone here...

Reply:
TaskCell can be dropped inside method `spawn`, which can make self invalid.